### PR TITLE
"Successful/fast 10M record upload"

### PR DESCRIPTION
**Pull request by skorpion-g**
https://trello.com/c/zi8ansUJ/55-optimized-generation-seeding-to-db
Notable changes in:
database-mysql/index.js
seeder/seeder,js
seeder/index.js

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,4 @@
 node_modules
 bundle.js
-product.csv
-review.csv
\ No newline at end of file
+product
+review
```
Comment (theofficetroll) [inline .gitignore:4]: Please add linebreak at the end.

File: database-mysql/index.js [hunk 2]
```diff
@@ -92,7 +92,7 @@ const create = (review) => {
 };
 
 const read = (id) => {
-  const readOne = 'SELECT * FROM review WHERE product_id = ?';
+  const readOne = 'SELECT * FROM review WHERE product_id = ?';z
```
Comment (theofficetroll) [inline database-mysql/index.js]: I assume the z is a typo?

File: package.json [hunk 3]
```diff
@@ -10,6 +10,7 @@
     "server": "node server/index.js",
     "test": "mocha './test/backend_spec/*.spec.js' --exit || true",
     "test-client": "mocha --r mock-css-modules --r @babel/register --r ./test/client_spec/helpers.js --r ./test/client_spec/dom.js './test/client_spec/*.spec.js' --exit || true",
+    "generate": "node seeder/seeder.js",
```
Comment (theofficetroll) [inline package.json:13]: Makes sense.

File: react-client/src/app.jsx [hunk 4]
```diff
@@ -34,7 +34,7 @@ class ReviewsModule extends React.Component {
     Modal.setAppElement('body');
 
     $.ajax({
-      url: `http://localhost:3003/api/reviews/2`,
+      url: `http://localhost:3003/api/reviews/${productID}`,
```
Comment (theofficetroll) [inline react-client/src/app.jsx:37]: Looks good.

File: schema.sql [hunk 5]
```diff
@@ -37,14 +37,14 @@ CREATE TABLE products (
  *  to create the database and the tables.*/
 /*
 to load csv's
-LOAD DATA LOCAL INFILE 'C:/Users/Frnkai/Test/SDC/service-reviews/CSV/product.csv'
+LOAD DATA LOCAL INFILE 'C:/Users/Frnkai/Test/SDC/service-reviews/CSV/product/product.csv'
```
Comment (theofficetroll) [inline schema.sql:40]: Organization change only, it appears.

File: schema.sql [hunk 6]
```diff
@@ -37,14 +37,14 @@ CREATE TABLE products (
...
 INTO TABLE products
 FIELDS TERMINATED BY ','
 LINES TERMINATED BY '\n'
 IGNORE 1 ROWS
 (id, product_id, product_name, comfort_average, durability_average, size_average, stars_average)
 
-LOAD DATA LOCAL INFILE 'C:/Users/Frnkai/Test/SDC/service-reviews/CSV/review.csv'
+LOAD DATA LOCAL INFILE 'C:/Users/Frnkai/Test/SDC/service-reviews/CSV/review/review.csv'
```
Comment (theofficetroll) [inline schema.sql:47]: Organization change only, it appears.

File: seeder/index.js [hunk 7]
```diff
@@ -4,14 +4,10 @@ const db = require('../database-mysql');
...
-const dates = ['Aug 08, 2020', 'Jan 01, 2020', 'Feb 12, 2020', 'Mar 22, 2020', 'Sep 11, 2020', 'Jul 02, 2020', 'Oct 29, 2020', 'Mar 10, 2020', 'Apr 05, 2020', 'Jun 03, 2020', 'Jun 05, 2020', 'May 05, 2020', 'May 25, 2020'];
-const seeder = require('../seeder/seeder')
-const { insertProducts, insertReviews } = require('../database-mysql/index');
-
-seeder.genProducts();
-seeder.genReviews(names, dates);
+const dbFunc = require('../database-mysql/index');
 
+dbFunc.uploadProdCSV();
+dbFunc.uploadRevCSV();
```
Comment (theofficetroll) [inline seeder/index.js:10]: Makes sense.

File: seeder/seeder.js [hunk 8]
```diff
@@ -4,66 +4,73 @@
...
+        product_id: i,
+        product_name: 'Placeholder',
+        comfort_average: helper.randomRating(),
+        durability_average: helper.randomRating(),
+        size_average: helper.randomRating(),
+        stars_average: helper.randomRating()
+      }
+      if (!writer.write(dataOne)) {
+        await new Promise((resolve) => writer.once('drain', resolve));
+      }
```
Comment (theofficetroll) [inline seeder/seeder.js:30]: Implemented Martin's method. Good good.

File: seeder/seeder.js [hunk 9]
```diff
@@ -4,66 +4,73 @@
...
+  return 'done';
 }
 
 const genReviews = async (names, dates) => {
   const writer = csvWriter();
+  let totalProd = 0;
+  for (var i = 1; i < 11; i++) {
+    writer.pipe(
+      fs.createWriteStream('./CSV/review/review.csv' + i, { flags: 'a' })
+    )
```
Comment (theofficetroll) [inline seeder/seeder.js:42]: Looks like you're breaking up the csv into multiple files to prevent errors.

File: seeder/seeder.js [hunk 10]
```diff
@@ -4,66 +4,73 @@
...
+          created_at: helper.randomDate(dates),
+          size_rating: helper.randomRating(),
+          durability_rating: helper.randomRating(),
+          comfort_rating: helper.randomRating(),
+          stars: helper.randomRating()
+        };
+        if (!writer.write(dataTwo)) {
+          await new Promise((resolve) => writer.once('drain', resolve));
+        }
+      }
```
Comment (theofficetroll) [inline seeder/seeder.js:65]: This all makes sense.
Pulling the total records to start off where the last left off.
Using the async/ await to ensure write is done.

File: seeder/seeder.js [hunk 11]
```diff
@@ -4,66 +4,73 @@
...
-module.exports = {
-  genProducts,
-  genReviews
-}
+const names = ['Bob', 'Charles', 'Jack', 'Jill', 'DrNy', 'Chris', 'Sarah', 'Kim', 'Mary', 'xxAio', 'giant', 'F.O', 'M.A', 'W.W', 'L.L', 'Arkoo', 'Mark', 'Lauren', 'Young', 'Rebecca'];
+const dates = ['Aug 08, 2020', 'Jan 01, 2020', 'Feb 12, 2020', 'Mar 22, 2020', 'Sep 11, 2020', 'Jul 02, 2020', 'Oct 29, 2020', 'Mar 10, 2020', 'Apr 05, 2020', 'Jun 03, 2020', 'Jun 05, 2020', 'May 05, 2020', 'May 25, 2020'];
+
+genProducts();
+genReviews(names, dates);
+
```
Comment (theofficetroll) [inline seeder/seeder.js:76]: Makes sense.

**Review by theofficetroll — APPROVED** [review]
Some minor stylistic notes.
For the most part makes sense.
Might want to complicate data later or expand word buckets, but this should be good at this point.
It may be less error prone to find a method of generating without splitting up the csv file, but this should be perfectly functional.

File: database-mysql/index.js (hunk 2 above)
Comment (skorpion-g) [reply database-mysql/index.js]: YES, typo hahahaha

File: .gitignore (hunk 1 above)
Comment (skorpion-g) [reply .gitignore:4]: This actually ignores all the directories that the generation script creates.

**Pull request merged**